test/test.pluggableauthhandler.ts

```diff
@@ -0,0 +1,455 @@
+} from '../src/auth/pluggable-auth-handler';
+import * as assert from 'assert';
+import {ExecutableError} from '../src/auth/pluggable-auth-client';
+
+const SAML_SUBJECT_TOKEN_TYPE = 'urn:ietf:params:oauth:token-type:saml2';
+
+describe('PluggableAuthHandler', () => {
+  const defaultHandlerOptions = {
+    command: './command/path/file.exe -opt',
+    outputFile: 'output',
+    timeoutMillis: 1000,
+  } as PluggableAuthHandlerOptions;
+
+  describe('Constructor', () => {
+    it('should not throw error with valid options', () => {
+      const client = new PluggableAuthHandler(defaultHandlerOptions);
+
+      assert(client instanceof PluggableAuthHandler);
+    });
+
+    it('should throw when options is missing command', () => {
+      const expectedError = new Error('No command provided.');
+      const invalidOptions = {
+        outputFile: 'output.txt',
+        timeoutMillis: 1000,
+      };
+
+      assert.throws(() => {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        return new PluggableAuthHandler(invalidOptions);
+      }, expectedError);
+    });
+
+    it('should throw when options is missing timeoutMillis', () => {
+      const expectedError = new Error('No timeoutMillis provided.');
+      const invalidOptions = {
+        command: './command -opt',
+        outputFile: 'output.txt',
+      };
+
+      assert.throws(() => {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        return new PluggableAuthHandler(invalidOptions);
+      }, expectedError);
+    });
+
+    it('should throw when command cannot be parsed', async () => {
+      const expectedError = new Error(
+        'Provided command: " " could not be parsed.'
+      );
+      const invalidHandlerOptions = {
+        command: ' ',
+        timeoutMillis: 10000,
+      } as PluggableAuthHandlerOptions;
+
+      assert.throws(() => {
+        return new PluggableAuthHandler(invalidHandlerOptions);
+      }, expectedError);
+    });
+  });
+
+  describe('RetrieveResponseFromExecutable', () => {
+    const sandbox = sinon.createSandbox();
+    let clock: sinon.SinonFakeTimers;
+    const referenceTime = Date.now();
+    let spawnEvent: child_process.ChildProcess;
+    let spawnStub: sinon.SinonStub<
+      [
+        command: string,
+        args: readonly string[],
+        options: child_process.SpawnOptions
+      ],
+      child_process.ChildProcess
+    >;
+    let defaultResponseJson: ExecutableResponseJson;
+    const expectedEnvMap = new Map();
+    expectedEnvMap.set(
+      'GOOGLE_EXTERNAL_ACCOUNT_TOKEN_TYPE',
+      SAML_SUBJECT_TOKEN_TYPE
+    );
+    expectedEnvMap.set('GOOGLE_EXTERNAL_ACCOUNT_INTERACTIVE', '0');
+    const expectedOpts = {
+      env: {
+        ...process.env,
+        ...Object.fromEntries(expectedEnvMap),
+      },
+    };
+
+    beforeEach(() => {
+      // Stub environment variables and set Allow Executables environment variable to 1
```

Comment:
// Stub environment variables and set GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES variable to 1.